### PR TITLE
Org Dashboard and Add Coupon Updates

### PR DESCRIPTION
**Pull request by dupreesi**
#### Issues this PR relates to
#704 #703 #706 
---

#### Any info the reviewer needs to know to test 
e.g. Add env vars, run npm install
- [ ] re-run dev data build
---

#### Any info the team needs to consider before merged into development
e.g. Will a database need to be rebuilt (and is this ok if existing data), add env vars

---

#### Quick Pull Request Checklist
Please make sure you ensure the following has been done before marking your PR as awaiting-review. For full checklist and process review, please refer to Process Guide (#8)
- Branch is up to date with develop
- Project runs
- You can download and install locally, and run the project locally
- All existing and new features run without bugs
- Checked that features/design works across all agreed browsers + devices
- Linter is used
- Code is consistent with agreed team style
- Naming of files is consistent with project 
- Tests all pass
- No console logs

### CODE REVIEW

File: client/src/components/Pages/OrgDashboard/AddCoupons/index.js [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+    message,
+    multiDateRange,
+    discountRate,
+    couponPrice,
+    code,
+  } = state;
+
+  const getAccount = async () => {
+    setBalanceLoading(true);
+    const { data } = await axios.get(API_ACCOUNT_URL);
```
Comment (thejoefriel) [inline client/src/components/Pages/OrgDashboard/AddCoupons/index.js:90]: should this be wrapped into a try and catch so we can handle errors? Or do you catch it in the later function it's included in further down?

File: client/src/helpers.js [hunk 2]
```diff
@@ -100,6 +103,42 @@ export const calculatePriceByRange = range => {
...
+ * this will discount the 14 free days and add 3 days to either side of internship start and end date
+ * @param {import("moment-range").MomentRange} range
+ */
+export const calculateCouponPriceByRange = (
+  startDate,
+  endDate,
+  discountRate,
+) => {
+  const range = moment.range(
+    moment(startDate).subtract(3, 'day'),
```
Comment (thejoefriel) [inline client/src/helpers.js:117]: One thing to be wary of here is I'm pretty sure .add and .subtract mutate the original date as well so you usually should do clone() first. It doesn't seem to be causing a problem so potentially moment-range protects from that?

File: server/controllers/coupon/getCoupons.js [hunk 3]
```diff
@@ -21,12 +21,23 @@ module.exports = async (req, res, next) => {
...
     }
     const coupon = await getCouponsQuery(query, queryProject).exec();
     if (!coupon[0]) {
       return next(boom.notFound());
     }
+
+    // check if coupon has been used by another user
+    if (coupon[0].intern && coupon[0].intern.toString() !== userId.toString()) {
+      return next(
+        boom.badRequest('This coupon has already been used by another user.'),
```
Comment (thejoefriel) [inline server/controllers/coupon/getCoupons.js:37]: Maybe make this error message the same as the one in teh following file to make consistent? i.e. This coupon has already been used by a user. If it was you please log in first and try again.

File: server/services/mailing/templates/organisation/couponCreated.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  return `
+    ${greeting(`Hi ${name}, `)}
+    ${content(`${organisation} has created a coupon for you.`)}
+    ${couponDetails.message.length &&
+      `Message: ${content(`${couponDetails.message}`)}`}
+    ${content(
+      `Enter ${couponDetails.code} and get ${couponDetails.discountRate} % off your next booking!`,
+    )}
+    ${content(
+      `Please make sure to be using your coupon during these dates: ${couponDetails.startDate} - ${couponDetails.endDate}!`,
```
Comment (thejoefriel) [inline server/services/mailing/templates/organisation/couponCreated.js:15]: yeah here just format the dates so they read a little nicer

File: server/controllers/coupon/getCoupons.js (hunk 3 above)
Comment (dupreesi) [reply server/controllers/coupon/getCoupons.js:37]: for this case the user is already logged in tho. the other error message is in the soft check when someone isn't logged in

Comment (thejoefriel) [reply server/controllers/coupon/getCoupons.js:37]: ok cool

**Pull request merged**